blog: trusted publishing growth, glsm, pending tp for orgs (#19019)
pypa/warehouse: docs/blog/posts/2025-11-07-trusted-publishers-coming-to-orgs.md

    content: "@miketheman@hachyderm.io"
---

## Summary

Trusted Publishing has proven popular since [its launch in 2023](2023-04-20-introducing-trusted-publishers.md).

**Recap:** Trusted Publishing enable software build platforms to publish packages to PyPI on your behalf, 
eliminating the need to manage long-lived authentication tokens.
After a one-time setup where you delegate publishing authority to your platform, 
it automatically obtains short-lived, scoped tokens for each build—no manual token management required.

Read the [Security Model](https://docs.pypi.org/trusted-publishers/security-model/)
for a deeper understanding of how Trusted Publishing works.

## Growing Adoption and Impact

Since its inception, Trusted Publishing has been adopted by communities and companies alike,
with ~45,000 projects configured for Trusted Publishing on PyPI so far.
In early 2024 we added the ability to track whether each file upload was done via a Trusted Publisher,
enabling us to analyze its impact over time.
Also in 2024, more platforms added were added to PyPI Trusted Publishing,
as detailed in our [April 2024 blog post](2024-04-17-expanding-trusted-publisher-support.md).

> <!-- more -->

Author: > we don't do this elsewhere, ...

5 of the more recent blogs have a Summary header, but point taken for this one, it doesn't add much to the content, will remove.

Hopefully posts like this and increased visibility in how to adopt Trusted Publishing
will help continue this growth trend.
Having folks like [Sviatoslav Sydorenko](https://github.com/webknjaz) and other contributors voluntarily maintain
the [widely-used GitHub Actions workflow for Trusted Publishing](https://github.com/marketplace/actions/pypi-publish)
that makes everything that much easier has been a huge help, support them as well if you can!
Using that action adds [Digital Attestations](2024-11-14-pypi-now-supports-digital-attestations.md) automatically. :lock:

I hope that when we examine these stats next year, we'll see even more growth
as more organizations adopt Trusted Publishing for their package publishing workflows.

Which brings me to...

## Expansion of Trusted Publishing to GitLab Self-Managed Beta

Since folks have been loving Trusted Publishing,
I'm excited to share that it's now available in beta for GitLab Self-Managed instances.
This means organizations running their own GitLab can now use Trusted Publishers
to publish packages more securely, without dealing with long-lived tokens.

> As a reminder, GitLab.com (the public instance) support has been around since early 2024.

Reviewer: ```suggestion
As a reminder, trusted publishing support for public GitLab.com instances has been available since early 2024.
```
